exec/src/exec.ml

```diff
@@ -21,39 +21,29 @@
 open Constr
 open Query
 open Utils
-open Coqstr
+open Interfaces
 
 let reduce_all env evm trm =
   EConstr.to_constr evm (Reductionops.nf_all env evm (EConstr.of_constr trm))
 
 let exec_request instr_t instr_trm func_trm =
-  match kind instr_t with
-  | Ind (i, _)
-    -> if Ind.ConsoleI.ref_is i
-       then let (instr_trm, args) = app_full instr_trm in
-            (match kind instr_trm with
-             | Construct (c, _)
-               -> let res =
-                    match (Ind.ConsoleI.constructor_of c, args) with
-                    | (Some Scan_console, [])
-                      -> coqstr_of_str (Console.scan ())
-                    | (Some Echo_console, [str])
-                      -> Console.echo (str_of_coqstr str);
-                         Ind.Unit.mkConstructor "tt"
-                    | _
-                      -> raise (Anomaly "Unknown [Console.i] constructor")
-                  in mkApp (func_trm, Array.of_list [res])
-             | _ -> raise (UnsupportedTerm "Trying to execute an axiomatic [Console.i]"))
-       else raise UnsupportedInterface
-  | _ -> raise UnsupportedInterface
+  let (instr_trm, args) = app_full instr_trm in
+  match kind instr_trm with
+  | Construct (c, _) ->
+     let res = primitive_semantic c args in
```

Comment:
I guess `primitive_semantic` raises an `UnsupportedInterface` when the constructor `c` is unknown?

Reply:
Add a comment about that in be538b1.

Reply:
Sounds good to me